openssl/aes.go

```diff
@@ -88,20 +85,14 @@ func (c *aesCipher) Encrypt(dst, src []byte) {
 	}
 
 	if c.enc_ctx == nil {
-		c.enc_ctx = C._goboringcrypto_EVP_CIPHER_CTX_new()
-		if c.enc_ctx == nil {
-			panic("cipher: unable to create EVP cipher ctx")
-		}
-
-		k := (*C.uchar)(unsafe.Pointer(&c.key[0]))
-
-		if C.int(1) != C._goboringcrypto_EVP_CipherInit_ex(c.enc_ctx, c.cipher, nil, k, nil, C.GO_AES_ENCRYPT) {
-			panic("cipher: unable to initialize EVP cipher ctx")
+		var err error
+		c.enc_ctx, err = newCipherCtx(c.cipher, C.GO_AES_ENCRYPT, c.key, nil)
+		if err != nil {
+			panic(err)
 		}
 	}
 
-	outlen := C.int(0)
-	C._goboringcrypto_EVP_CipherUpdate(c.enc_ctx, (*C.uchar)(unsafe.Pointer(&dst[0])), &outlen, (*C.uchar)(unsafe.Pointer(&src[0])), C.int(aesBlockSize))
+	C.go_openssl_EVP_EncryptUpdate_wrapper(c.enc_ctx, base(dst), base(src), aesBlockSize)
```

Comment:
Would it make sense to check the return value?

Reply:
It does. We probably did not check the return because upstream Go is not checking in neither:

https://github.com/golang/go/blob/81efd7b347dd6d7f12fd49c6eee0274005734c71/src/crypto/internal/boring/aes.go#L186

BoringSSL [AES_ctr128_encrypt](https://commondatastorage.googleapis.com/chromium-boringssl-docs/aes.h.html#AES_ctr128_encrypt) does not return any value, so upstream has nothing to check. OpenSSL counterpart does return a status code, so we should be checking it.